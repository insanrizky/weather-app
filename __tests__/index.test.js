import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

import App from '../pages/index.js'
const mockStore = configureMockStore();
const store = mockStore({
  weather: {
    current: {}
  }
});

describe('With Enzyme', () => {
  it('App show main page', () => {
    const app = shallow(<App />)
    expect(app.find('div')).toBeTruthy();
  })
});

describe('With Snapshot Testing', () => {
  it('App show main page', () => {
    const component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
});
