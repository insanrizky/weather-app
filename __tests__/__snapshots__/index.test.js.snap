// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App show main page 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-ocean-dark m-b-20"
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand"
        href="#"
      >
        <img
          alt="logo"
          src="/static/favicon.png"
        />
         
        <span>
          Weather App
        </span>
      </a>
      <button
        aria-controls="navbarText"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-target="#navbarText"
        data-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarText"
      >
        <ul
          className="navbar-nav mr-auto mt-2 mt-lg-0"
        />
        <ul
          className="navbar-nav my-2 my-lg-0"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www.linkedin.com/in/muhamad-insan-rizky/"
            >
              <span
                className="fab fa-linkedin-in"
              />
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://github.com/insanrizky/weather-app"
            >
              <span
                className="fab fa-github"
              />
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div
      className="row justify-content-md-center m-50"
    >
      <div
        className="col-md-8"
      >
        <form
          id="form-search"
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Type a city (ex: Jakarta, London, Yokohama, etc)"
              type="text"
              value=""
            />
          </div>
        </form>
      </div>
      <div
        className="col-md-2"
      >
        <button
          className="btn btn-success btn-block"
          form="form-search"
          type="submit"
        >
          <span
            className="fas fa-sun"
          />
           Get Weather!
        </button>
      </div>
    </div>
    <hr />
    <div>
      <div
        className="row"
      >
        <div
          className="col-md-3"
        >
          <div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <div
                  className=""
                >
                  <div
                    className="text-center"
                  >
                    <img
                      className="weather__icon--big"
                    />
                    <p />
                  </div>
                  <hr />
                  <div
                    className=""
                  >
                    <h6>
                      Tue, Feb 26
                    </h6>
                    <h5>
                      , 
                    </h5>
                    <h4>
                      ℃ / 
                      ℉
                    </h4>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-md-9"
        >
          <div
            className="card m-b-20"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Today's Weather
              </h5>
              <hr />
              <div
                className="row"
              />
            </div>
          </div>
          <div
            className="card m-b-20"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Weather Forecast
              </h5>
              <hr />
              <div
                className="row"
              />
            </div>
            <div
              aria-hidden="true"
              aria-labelledby="exampleModalLabel"
              className="modal fade "
              id="exampleModal"
              role="dialog"
              tabIndex="-1"
            >
              <div
                className="modal-dialog"
                role="document"
              >
                <div
                  className="modal-content"
                >
                  <div
                    className="modal-header"
                  >
                    <h5
                      className="modal-title"
                      id="exampleModalLabel"
                    >
                      Forecast Detail
                    </h5>
                    <button
                      aria-label="Close"
                      className="close"
                      data-dismiss="modal"
                      onClick={[Function]}
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        ×
                      </span>
                    </button>
                  </div>
                  <div
                    className="modal-body"
                  >
                    
                  </div>
                  <div
                    className="modal-footer"
                  >
                    <button
                      className="btn btn-sm btn__outline--ocean"
                      data-dismiss="modal"
                      onClick={[Function]}
                      type="button"
                    >
                      Tutup
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="footer"
  >
    Copyright © 2019 All rights reserved | Powered by: 
    <a
      href="https://www.apixu.com"
    >
      APIXU
    </a>
  </div>
</div>
`;
